app/decorators/collection_decorator.rb

```diff
@@ -0,0 +1,43 @@
+class CollectionDecorator < SimpleDelegator
+
+  attr_reader :data
+
+  # Pseuds need to be decorated with various stats from the "_source" when
+  # viewing search results, so we first load the pseuds with the base search
+  # class, and then decorate them with the data.
+  def self.load_from_elasticsearch(hits)
+    items = Collection.load_from_elasticsearch(hits)
+    decorate_from_search(items, hits)
+  end
+
+  def self.decorate_from_search(results, search_hits)
```

Comment:
Can you keep the comment `# TODO: pull this out into a reusable module` on this method, similar to the pseud version?